### ISSUE TITLE
consolidate SslStream to single file

### ISSUE THREAD

**Issue by wfurt** [dotnet/runtime#69045]
This was suggested by @stephentoub on #68678. The current split is somewhat artificial  because of historical reasons. 
There are two challenges: 
- preserving history & blame. That should be possible AFAIK but it should be tested.
- We will need to make some bigger refactor of UnitTests. They link SslStream.cs now and that have fakes for the rest. We would either need to compile some parts conditionally, refactor the tests of find some other way how to get similar functionality. 

We may also add section tags so at least in VS related sections are collapsible.

**Comment by stephentoub** [comment]
> preserving history & blame

This isn't being negatively impacted by your current PR?

**Comment by wfurt** [comment]
Blame works because with git move it is essentially still the same file. History does not seems to. Maybe something we can still fix.

**Comment by wfurt** [comment]
No. I was using GitHub UI. I know we had some tooling to replay history when we moved to new repo. Perhaps @ViktorHofer has some more insight.

**Comment by ViktorHofer** [comment]
Rewriting history requires to force push a branch and we obviously can't do that anymore. `git log --follow` usually works fine for these kinds of changes.

**Comment by rzikm** [comment]
We don't want to lose access to git blame on the individual files, so we will not be consolidating the source files.
